crates/chia-bls/src/cached_bls.rs

```diff
@@ -1,253 +1,260 @@
 }
 
-impl Default for BLSCache {
+impl Default for BlsCache {
     fn default() -> Self {
-        Self::new(None)
+        Self::new(NonZeroUsize::new(50000).unwrap())
     }
 }
 
-impl BLSCache {
-    pub fn new(cache_size: Option<NonZeroUsize>) -> BLSCache {
-        let cache: LruCache<[u8; 32], GTElement> = LruCache::new(
-            cache_size.unwrap_or(NonZeroUsize::new(50000).expect("50000 should be non-zero")),
-        );
-        Self { cache }
+impl BlsCache {
+    pub fn new(cache_size: NonZeroUsize) -> Self {
+        Self {
+            cache: LruCache::new(cache_size),
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.cache.len()
     }
 
-    pub fn aggregate_verify<
-        M: IntoIterator<Item = T>,
-        T: AsRef<[u8]>,
-        P: IntoIterator<Item = U>,
-        U: Borrow<PublicKey>,
-    >(
+    pub fn is_empty(&self) -> bool {
+        self.cache.is_empty()
+    }
+
+    pub fn aggregate_verify(
         &mut self,
-        pks: P,
-        msgs: M,
+        pks: impl IntoIterator<Item = impl Borrow<PublicKey>>,
+        msgs: impl IntoIterator<Item = impl AsRef<[u8]>>,
         sig: &Signature,
     ) -> bool {
         let iter = pks.into_iter().zip(msgs).map(|(pk, msg)| -> GTElement {
+            // Hash pubkey + message
             let mut hasher = Sha256::new();
             hasher.update(pk.borrow().to_bytes());
-            hasher.update(msg.as_ref()); // pk + msg
-            let h: [u8; 32] = hasher.finalize().into();
+            hasher.update(msg.as_ref());
+            let hash: [u8; 32] = hasher.finalize().into();
 
-            if let Some(pairing) = self.cache.get(&h).cloned() {
-                // equivalent to `if pairing is not None`
+            // If the pairing is in the cache, we don't need to recalculate it.
+            if let Some(pairing) = self.cache.get(&hash).cloned() {
                 return pairing;
             }
-            // if pairing is None then make pairing and add to cache
+
+            // Otherwise, we need to calculate the pairing and add it to the cache.
             let mut aug_msg = pk.borrow().to_bytes().to_vec();
-            aug_msg.extend_from_slice(msg.as_ref()); // pk + msg
-            let aug_hash: Signature = hash_to_g2(&aug_msg);
+            aug_msg.extend_from_slice(msg.as_ref());
+            let aug_hash = hash_to_g2(&aug_msg);
 
-            let pairing: GTElement = aug_hash.pair(pk.borrow());
             let mut hasher = Sha256::new();
             hasher.update(&aug_msg);
-            let h: [u8; 32] = hasher.finalize().into();
-            self.cache.put(h, pairing.clone());
+            let hash: [u8; 32] = hasher.finalize().into();
+
+            let pairing = aug_hash.pair(pk.borrow());
+            self.cache.put(hash, pairing.clone());
             pairing
         });
-        agg_ver_gt(sig, iter)
+
+        aggregate_verify_gt(sig, iter)
     }
 }
 
-// Python Functions
+#[cfg(feature = "py-bindings")]
```

Comment:
This is a new thing I'm going to be moving our code to if it works out - it keeps all of the imports contained into a Python binding module rather than repeating `#[cfg(feature = "py-bindings")]` over and over.